aptos-move/framework/aptos-framework/sources/native_bridge.move

```diff
@@ -0,0 +1,816 @@
+        // Create bridge transfer details  
+        let details = native_bridge_store::create_details(  
+            initiator_address,  
+            ethereum_address, 
+            amount,  
+            nonce  
+        );  
+    
+        // Generate a unique bridge transfer ID  
+        // Todo: pass the nonce in here and modify the function to take a nonce. Or only use the nonce in native_bridge_store
+        let bridge_transfer_id = native_bridge_store::bridge_transfer_id(&details);  
+    
+        // Add the transfer details to storage  
+        native_bridge_store::add(nonce, details);
+
+        // Burn the amount from the initiator  
+        native_bridge_core::burn(initiator_address, amount);  
+    
+        let bridge_events = borrow_global_mut<BridgeEvents>(@aptos_framework);
+
+        // Emit an event with nonce  
+        event::emit_event(  
+             &mut bridge_events.bridge_transfer_initiated_events,
+            BridgeTransferInitiatedEvent {  
+                bridge_transfer_id,  
+                initiator: initiator_address,  
+                recipient,  
+                amount,  
+                nonce,  
+            }  
+        );  
+    }
+
+    /// Completes a bridge transfer by the initiator.
+    ///  
+    /// @param caller The signer representing the bridge relayer.  
+    /// @param initiator The initiator's Ethereum address as a vector of bytes.  
+    /// @param bridge_transfer_id The unique identifier for the bridge transfer.  
+    /// @param recipient The address of the recipient on the Aptos blockchain.  
+    /// @param amount The amount of assets to be locked.  
+    /// @param nonce The unique nonce for the transfer.    
+    /// @abort If the caller is not the bridge relayer or the transfer has already been processed.  
+    public entry fun complete_bridge_transfer(  
+        caller: &signer,  
+        bridge_transfer_id: vector<u8>,
+        initiator: vector<u8>,  
+        recipient: address,  
+        amount: u64,  
+        nonce: u64  
+    ) acquires BridgeEvents {  
+        // Ensure the caller is the bridge relayer
+        native_bridge_configuration::assert_is_caller_relayer(caller);  
+
+        // Check if the bridge transfer ID is already associated with an incoming nonce
+        let incoming_nonce_exists = native_bridge_store::is_incoming_nonce_set(bridge_transfer_id);
+        assert!(!incoming_nonce_exists, ETRANSFER_ALREADY_PROCESSED); // Abort if the transfer is already processed
+
+        let ethereum_address = ethereum::ethereum_address_no_eip55(initiator);
+
+        // Validate the bridge_transfer_id by reconstructing the hash
+        let combined_bytes = vector::empty<u8>();
+        vector::append(&mut combined_bytes, bcs::to_bytes(&initiator));
+        vector::append(&mut combined_bytes, bcs::to_bytes(&recipient));
+        vector::append(&mut combined_bytes, bcs::to_bytes(&amount));
+        vector::append(&mut combined_bytes, bcs::to_bytes(&nonce));
```

Comment:
think this can use the function bridge_transfer_id() instead

Reply:
It would need to be done with 

```
        // Create bridge transfer details  
        let details = native_bridge_store::create_details(  
            initiator_address,  
            ethereum_address, 
            amount,  
            nonce  
        );  
    
        // Generate a unique bridge transfer ID  
        let bridge_transfer_id = native_bridge_store::bridge_transfer_id(&details);  
```

You don't think it simpler like it is?

Reply:
oh yeah, true.